gui/wxpython/gui_core/pyedit.py

```diff
@@ -368,6 +376,10 @@ def OnSave(self, event):
         else:
             self.SaveAs()
 
+        self.recent_files.AddFileToHistory(
+            filename=self.filename,
+        )
```

Comment:
SaveAs allows user to cancel the operation, so that should now write the file to history. Save does not seems to be involved in temp files, but it seems to be response to user action on saving in both cases (Save and Save as), so perhaps it is a better candidate for this call than OnSave. An alternative would be to check self.filename and self.tempfile before calling AddFileToHistory in OnSave (to see the result of SaveAs).

Reply:
All right, I fixed it.